datafusion/sqllogictest/README.md

```diff
@@ -177,14 +177,32 @@ You can update the tests / generate expected output by passing the `--complete`
 cargo test --test sqllogictests -- ddl --complete
 ```
 
-#### sqllogictests
+#### Running tests: `scratchdir`
 
-sqllogictest is a program originally written for SQLite to verify the correctness of SQL queries against the SQLite
-engine. The program is engine-agnostic and can parse sqllogictest files (`.slt`), runs queries against an SQL engine and
-compare the output to the expected output.
+The DataFusion sqllogictest runner automatically creates a directory
+named `test_files/scratch/<filename>`, creating it if needed and
+clearing any file contents if it exists.
 
-Tests in the `.slt` file are a sequence of query record generally starting with `CREATE` statements to populate tables
-and then further queries to test the populated data (arrow-datafusion exception).
+For example, the `test_files/copy.slt` file should use scratch
```

Comment:
This looks great to me! I ran into some issues needing to make sure each table had a unique subpath betwen copy/insert tests in #7294, and this would simplify that.

Reply:
Merged https://github.com/apache/arrow-datafusion/pull/7294 and I updated this PR with the changes from there